pandas/core/ops.py

```diff
@@ -276,15 +276,36 @@ def __init__(self, left, right, name):
 
         self.left = left
         self.right = right
-        lvalues = self._convert_to_array(left, name=name)
-        rvalues = self._convert_to_array(right, name=name, other=lvalues)
 
-        self.is_timedelta_lhs = is_timedelta64_dtype(left)
-        self.is_datetime_lhs = is_datetime64_dtype(left)
-        self.is_integer_lhs = left.dtype.kind in ['i', 'u']
-        self.is_datetime_rhs = is_datetime64_dtype(rvalues)
-        self.is_timedelta_rhs = is_timedelta64_dtype(rvalues)
-        self.is_integer_rhs = rvalues.dtype.kind in ('i', 'u')
+        self.is_offset_lhs = False
+        self.is_datetime_lhs = False
+        self.is_timedelta_lhs = False
+        self.is_integer_lhs = False
+        self.is_offset_rhs = False
+        self.is_datetime_rhs = False
+        self.is_timedelta_rhs = False
+        self.is_integer_rhs = False
+
+        # DateOffsets skip array conversion
+        if self._is_offset(left):
```

Comment:
why don't you add the variables: ``self.is_offset_rhs = self._is_offset(right)`` (and for the left), then just have ``self._convert_to_array(....)`` just return the data (w/o modification). I think a bit cleaner that way.